### ISSUE TITLE
[Android][Security] How to prevent Tapjacking

### ISSUE THREAD

**Issue by ShuheiSuzuki-07** [pichillilorenzo/flutter_inappwebview#2002]
- [x] I have read the [Getting Started](https://inappwebview.dev/docs/intro/) section
- [x] I have already searched for the same problem

## Environment

| Technology            | Version       |
|-----------------------| ------------- |
| Flutter version       |       3.3.10        |
| Plugin version        |       ~6.0.0       |
| Android version       |      Android 11 or lower      |
| iOS version           |        -       |
| macOS version         |      -         |
| Xcode version         |       -        |
| Google Chrome version |        -       |

Device information: 

## Description

**Expected behavior:** 
Inability to perform Tapjacking

**Current behavior:** 
Tapjacking is possible on InAppBrowser

Learn more about Tapjacking
https://developer.android.com/privacy-and-security/risks/tapjacking?hl=ja#risk_full_occlusion

## Steps to reproduce

1. launch the Tapjacking attack app (this app is not harmful).
2. Auto start example app
3. confirm that InAppBrowser can be operated under the Tapjacking app.

## Images   

![attack](https://github.com/pichillilorenzo/flutter_inappwebview/assets/118415919/4c6b8ae1-1057-476c-bcb8-8643ac6cc1e3)

## tapjacking demo app

Demo app is below
[tapjackingdemo.apk.zip](https://github.com/pichillilorenzo/flutter_inappwebview/files/14082120/tapjackingdemo.apk.zip)

The source code for demo app is below
[Tapjacking-ExportedActivity.zip](https://github.com/pichillilorenzo/flutter_inappwebview/files/14092393/Tapjacking-ExportedActivity.zip)

**Comment by pichillilorenzo** [comment]
Thanks, I didn’t know about that!
I will add some settings to enable/disable the specific View settings as written in the Android URL docs.

**Comment by ShuheiSuzuki-07** [comment]
@pichillilorenzo 
Thanks for the quick reply.
I am happy to be of service.
I will send you the APK file of the app to try Tapjacking in action, as well as the APK file of the app.
I have also attached the source code in case you are unsure about running the APK file.
Please do not hesitate to contact me.

**Comment by Avijaygovinda** [comment]
@ShuheiSuzuki-07 Hi, can you please share us, how to prevent from tapjacking, it will helpful for lots of flutter devolopers.

**Comment by ShuheiSuzuki-07** [comment]
@Avijaygovinda 

I am not a security expert but I will decompress.
For Flutter apps, just incorporate the following code into the onCreate method of MainActivity.kt.

```kotlin
    val view = findViewById<View>(android.R.id.content).rootView
    view.filterTouchesWhenObscured = true
```

But if you are using InAppBrowser in flutter_inappwebview it will work in a different Activity than MainActivity. Therefore, it is necessary for the creator of the plugin to take care of this. So I opened this issue.
If you want to prevent TapJacking only on MainActivity, you can use the method described above.

**Comment by Avijaygovinda** [comment]
@ShuheiSuzuki-07 sorry to disturb you, i have requirements to restrict from tap jacking in my flutter app,

```
class MainActivity: FlutterActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        
        // Retrieve the root view of the activity
        val rootView = findViewById<View>(android.R.id.content).rootView
        
        // Set the filterTouchesWhenObscured property to true
        rootView.filterTouchesWhenObscured = true
    }
} 
```


i used this code i think its not working as expected, bacause i checking with an app which is tapjacking window will appear on app, that window still apears when i added the above provided code, so please if you know any solutions provide with us.

**Comment by ShuheiSuzuki-07** [comment]
@Avijaygovinda 

Let me put the matter straight once and for all.

You put in the `rootView.filterTouchesWhenObscured = true` response. But my `tapjacking demo app` has been activated. Am I correct in my understanding that this is the case?

If this understanding is correct, it is not a problem that my `tapjacking demo app` is launched.

In case of a TapJacking attack, the `tapjacking demo app` I created will be launched, and furthermore, your app behind it can be manipulated.
If the countermeasure is working, my `tapjacking demo app` will be launched, but your app behind it will not be able to operate.

In other words, the TapJacking attack is transparent to the app behind the rogue app.

As a side note, the purpose of this issue is only to present the problem with this plugin. Please keep in mind that we don't all want things to go too far off-topic.
